MerlinAU.sh

```diff
@@ -1789,18 +1793,18 @@ _RunFirmwareUpdateNow_()
         return 1
     fi
 
-	# Use Get_Custom_Setting to retrieve the previous choice
-	previous_choice="$(Get_Custom_Setting "ROGBuild" "n")"
+    # Use Get_Custom_Setting to retrieve the previous choice
+    previous_choice="$(Get_Custom_Setting "ROGBuild" "n")"
 
-	# Navigate to the firmware directory
-	cd "$FW_BIN_DIR"
+    # Navigate to the firmware directory
+    cd "$FW_BIN_DIR"
 
-	# Detect ROG and pure firmware files
-	rog_file="$(ls | grep -i '_rog_')"
-	pure_file="$(ls | grep -iE '(\.w|\.pkgtb|\.trx)$' | grep -iv 'rog')"
+    # Detect ROG and pure firmware files
+    rog_file="$(ls | grep -i '_rog_')"
+    pure_file="$(ls -1 | grep -iE '.*[.](w|pkgtb|trx)$' | grep -iv 'rog')"
```

Comment:
The modified regular expression is just as precise as the previous one because it still explicitly requires the dot char **before** any of the three extensions **and** additionally & explicitly requires at least one character before the dot. Yeah, it may seem unnecessary but I've learned to expect the unexpected when it comes to filenames.